From 1d802b56c2fcea6f793cbaf2c0e99556c847ca4b Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Tue, 14 Sep 2010 08:30:10 +0000
Subject: [PATCH 0264/1092] cbus: remove device_release completion

We don't do anything before releasing the device
so we might as well return as soon as possible.

Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Tony Lindgren <tony@atomide.com>
---
 drivers/cbus/retu.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/cbus/retu.c b/drivers/cbus/retu.c
index bc81a9b..15368dc 100644
--- a/drivers/cbus/retu.c
+++ b/drivers/cbus/retu.c
@@ -56,8 +56,6 @@ static int retu_is_vilma;
 static struct tasklet_struct retu_tasklet;
 spinlock_t retu_lock = SPIN_LOCK_UNLOCKED;
 
-static struct completion device_release;
-
 struct retu_irq_handler_desc {
 	int (*func)(unsigned long);
 	unsigned long arg;
@@ -491,11 +489,6 @@ static int __devexit retu_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static void retu_device_release(struct device *dev)
-{
-	complete(&device_release);
-}
-
 static struct platform_driver retu_driver = {
 	.probe		= retu_probe,
 	.remove		= __devexit_p(retu_remove),
@@ -507,9 +500,6 @@ static struct platform_driver retu_driver = {
 static struct platform_device retu_device = {
 	.name		= "retu",
 	.id		= -1,
-	.dev = {
-		.release = retu_device_release,
-	}
 };
 
 /**
@@ -525,8 +515,6 @@ static int __init retu_init(void)
 		machine_is_nokia_n810() || machine_is_nokia_n810_wimax()))
 			return -ENODEV;
 
-	init_completion(&device_release);
-
 	ret = platform_driver_register(&retu_driver);
 	if (ret < 0)
 		return ret;
@@ -547,7 +535,6 @@ static void __exit retu_exit(void)
 {
 	platform_device_unregister(&retu_device);
 	platform_driver_unregister(&retu_driver);
-	wait_for_completion(&device_release);
 }
 
 subsys_initcall(retu_init);
-- 
1.7.11.2

